src/utils/child_process.rs

```diff
@@ -89,8 +91,135 @@ impl Nanny {
 }
 
 fn boot_desktop_file(path: &Path) -> std::io::Result<Child> {
-    let args = format!( "`if [ \"$(grep '^X-GNOME-Autostart-enabled' {:?} | tail -1 | sed 's/^X-GNOME-Autostart-enabled=//' | tr '[A-Z]' '[a-z]')\" != 'false' ]; then grep '^Exec' {:?} | tail -1 | sed 's/^Exec=//' | sed 's/%.//' | sed 's/^\"//g' | sed 's/\" *$//g'; else echo 'exit'; fi`", path , path);
-    Command::new("sh").arg("-c").arg(args).spawn()
+    use std::io::Error as StdError;
```

Comment:
Would you mind to double check if the errors you throw are in line with the error-message rework done by #472, that'd be great.
Thanks!

Reply:
emm, I don't get this.

You mean
1. use `thiserror`, not `ErrorKind`?
2. `error-message` should be distinguishing?